pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1057,7 +1057,12 @@ self: super:
               }
               + "/${name}";
             # 2025-04-09: jailbreak to allow bytestring >= 0.12, text >= 2.1
-            jailbreak = true;
+            # Note: jailbreak can't handle constraints under an if(flag)
+            postPatch = ''
+              sed -i ${name}.cabal \
```

Comment:
I have a weak preference for `substituteInPlace --replace-fail` since that would let us know when to remove this.

FTR, we intentionally leave conditionals alone https://github.com/NixOS/jailbreak-cabal/commit/99eac40deb481b185fd93fd307625369ff5e1ec0.

Reply:
A flag conditional may, besides restricting the version ranges, set any number of things including e.g. compilation flags. If the flag is marked as automatic, Cabal will attempt to automatically set it based on the preinstalled (by us) versions of packages. If we remove those constraints, it just has to go with the default value of the flag, which may mean incorrect settings and the build failing incorrectly, e.g. because a CPP var isn't set.

Reply:
> I have a weak preference for substituteInPlace --replace-fail since that would let us know when to remove this.

It's not possible with substituteInPlace: this code applies to 3 packages and the whitespace is different in each .cabal file, so you need a regex. However we can make sed fail too.